client/src/pages/admin/pages/data-sources.tsx

```diff
@@ -28,7 +29,10 @@ export const DataSources = () => {
   return (
     <>
       <h1 className="text-3xl font-bold">Data Sources</h1>
-      <div className="flex flex-wrap gap-4">
+      <Link to="/admin/add-data-source">
```

Comment:
I don't think it's needed to have the 'add data source' as it's own route/link

Reply:
Will sort this in the modal PR